DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/merge_sort/src/CMakeLists.txt

```diff
@@ -28,6 +28,21 @@ else()
     endif()
 endif()
 
+# Older quartus versions do not support as many kernels in SYCL HLS when compiling for hardware
+if (IS_BSP MATCHES "0")
+    execute_process(COMMAND quartus_sh --version
+        OUTPUT_VARIABLE QUARTUS_VERSION
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+    string(REGEX MATCH "[0-9][0-9]\.[0-9]\.[0-9]"
+        QUARTUS_VERSION
+        "${QUARTUS_VERSION}")
```

Comment:
Can you check if there was any match to begin with?
Customers may not have Quartus setup properly etc.
If there is no regex match then print out a message stating that the quartus version could not be determined and that the default values (large) is going to be used

Reply:
Good point, added a check